### ISSUE TITLE
onchange should not be called if state hasn't actually changed

### ISSUE THREAD

**Issue by timwis** [sethvincent/send-action#2]
Hi Seth! I brought this up in yoshuawuyts/choo#25, but I feel it's worth posting here too. The `onchange` handler suggests it is called on a _change_, but it's actually called every time `send()` is used, even if the state hasn't changed, even if no payload was provided. What do you think?

**Comment by yoshuawuyts** [comment]
Hmmm, despite the name this might be something that needs to be solved at the framework level. In `choo` there's the concept of `reducers` and `effects` - while it makes sense to re-render on a `reducer`, it doesn't always when an `effect` is called. So a simple optimization we could do is:

``` js
function onUpdate(newState, send)
  if (prevState !== newState) render(newState, send)
}
```

**edit:** lol, that's pretty much what you're saying in the linked post on the `choo` repo :sparkles:

**Comment by sethvincent** [comment]
Is there any downside for choo to putting `prevState !== newState` in send-action (or other possible downsides)?

@yoshuawuyts I'm seeing that you already added a check in choo to handle this.

The sad answer is that I probably just forgot to add that little if statement. Only rendering if state has actually changed seems like expected behavior.

**Comment by yoshuawuyts** [comment]
Yup, yeah I added this to choo - don't think there are any downsides to adding it there :sparkles: - does that resolve the issue? I think it does, right?

**Comment by sethvincent** [comment]
Added a check for changed state here: https://github.com/sethvincent/send-action/blob/master/index.js#L14-L16

published in v2.0.0 along with your recent pull requests @timwis. also added a link to choo in the readme!
